package.json

```diff
@@ -1,17 +1,20 @@
 {
   "repository": "https://github.com/kiltprotocol/sdk-js",
+  "name": "root-workspace",
   "private": true,
   "workspaces": [
     "packages/*",
     "docs/*"
   ],
   "license": "BSD-4-Clause",
   "scripts": {
-    "check": "tsc -p tsconfig.json --noEmit",
-    "build": "yarn workspaces foreach -p -t --exclude '{root-workspace}' run build",
+    "check": "tsc -b tsconfig.json",
+    "build": "yarn clean && yarn workspaces foreach -pt --exclude 'root-workspace' run build",
```

Comment:
If we run `yarn clean` before a build, we lose the incremental compilation benefit you mention in the PR description, I guess. So we should probably not clean by default.

Reply:
This is by design; `yarn build` is only meant to be run for production builds essentially. Running a clean beforehand makes sure we don't get any weird artefacts. For standard builds and rebuilds in development you can simply use `build:dev`.

Reply:
As mentioned below, we should then somehow advertise that `build:dev` is the to-go command then.